### PR TITLE
refactor: refactor key handler to be neater

### PR DESCRIPTION
**Pull request by joannalikespi**
Removed global key handler and replaced it with key handlers local to each component

### CODE REVIEW

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: I think `global as unknown` will also work

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: I tried changing it to (global as unknown).__static = '';  and it gave me errors - do you want me to hunt down a fix that will allow me to use unknown, or should i just leave it as any?

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Please rebase after #85 is merged and also modify the style of this test following that too.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: If you leave this out, you can specify the behavior separately. See #85

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Let's not suppress these warnings - they don't break the tests and it might be worthwhile to address those in the future.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [inline src/renderer/components/recording-page/recording-page.test.tsx]: Prefer `userEvent` to `fireEvent`

File: src/renderer/index.tsx
Comment (iluminar-yi) [inline src/renderer/index.tsx:8]: Can you also delete the files at `services/key-event-handler-registry` since they are no longer used?

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Please rebase after #85 is merged.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: I'm not sure is userEvent has the functionality needed to perform this test.  I need to simulate holding a key down for a few seconds and then releasing it - from what I can see, the closest thing userEvent has is allowing for text to be typed into a input or text element (rather than just typed without focus on a specific element).

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: I tried just using jest.mock('../../services/media'), but it wouldn't accept the mock unless I added in a function for the mock to use.  Should i set it so that the jest.mock call passes in an empty function initially, and then specify the behavior for each method later?

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: Should I hold off on pushing any commits to this branch until 85 is merged?

Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: Yes that's what I meant. And it's merged already, so you are free to rebase now.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: Sounds good.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: Show me the error that bugs `unknown`.
Also, look into setting the global variables for jest.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: It depends. Show me the exact error (after you rebase the branch).

If default mock doesn't work, then you can still do `switchOnAudioInput: jest.fn(),` and specify the behavior later.

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (joannalikespi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: image
image

**Review by iluminar-yi — APPROVED** [review]

File: src/renderer/components/recording-page/recording-page.test.tsx
Comment (iluminar-yi) [reply src/renderer/components/recording-page/recording-page.test.tsx]: Seems like it should still be set in jest config rather than in the test code because TypeScript doesn't like that.

**Comment by iluminar-yi** [comment]
Next time, before confirming the merge, make sure the generated commit message follows the commit convention. If not, you will need to manually fix it.

**Pull request merged**
